### ISSUE TITLE
Makumba should be usable out-of-the-box

### ISSUE THREAD

**Issue by ghost** [makumba/makumba#536]
**Reported by @manuelbernhardt on 24 Sep 2007 08:45 UTC**
Whilst we claim that Makumba is very simple to use, we don't address enough the initial configuration issue, which is a blocker for many amateur developers.

A simpler, more integrated solution than the one proposed in bug 985 would be to ship Makumba with an easy installation mechanism. The users would download a war which they'd simply drop in their application server (whose installation would be throughly documented), and then have access to a set-up interface in their browser, through which they could configure their database (we could also give support for an integrated DB, like the IBM java one) and start straight ahead trying out things (generation of a demo webapp etc).

basically, this would require to write a servlet / webapp able of generating Makumba configuration files and containing a bit of demonstration pages.

Migrated-From: http://trac.makumba.org/ticket/988

**Comment by ghost** [comment]
**Comment by @stefanb on 24 Sep 2007 14:54 UTC**
I agree that this would be more useful and flexible than virtual appliance (bug 985), but probably a bit more difficult to implement (not impossible!).

Installation script(s) would have to consider things like
* checking environment (various DBs, java versions, servlet containers, permissions, filesystems...)
* passing ant tests on the configured DB 
* writing config files
* serve sample web app for the user to test the UI layer

The "makumba configuration wizard" webapp could offer
* creating a new makumba webapp (new context, possibly with a sample webapp)
* makumbize an existing webapp (drop in jars + config files)

**Comment by ghost** [comment]
**Comment by @rudolfmayer on 24 Sep 2007 22:04 UTC**
(In reply to comment #1)
> I agree that this would be more useful and flexible than virtual appliance (bug
> 985), but probably a bit more difficult to implement (not impossible!).
> 
> Installation script(s) would have to consider things like
> * checking environment (various DBs, java versions, servlet containers,
> permissions, filesystems...)


if you say you provide support for this on e.g. debian and redhat, it should be rather easy to make a package that has dependencie to all the needed libs (even apache commons are available in debian as package, so it should be possible to find it for all..)

**Comment by ghost** [comment]
**Comment by @manuelbernhardt on 10 Mar 2010 10:23 UTC**
I am closing this bug as makumba can be now used (and is being successfuly used) out-of-the-box using the maven deployer.

Additional packaging / deployment solutions such as a debian package would be very cool to have. That is, this is no longer a blocker (you can very easily install maven on debian systems too and go through the quickstart).
